test/testpermissions.cpp

```diff
@@ -111,18 +116,58 @@ private slots:
 
+        const auto removeReadOnly = [&] (const QString &file)  {
+            auto fileInfoToDelete = QFileInfo(fakeFolder.localPath() + file);
+            QFile(fakeFolder.localPath() + file).setPermissions(QFile::WriteOwner | QFile::ReadOwner);
+            const auto readOnlyFolder = !static_cast<bool>(std::filesystem::status(fileInfoToDelete.absolutePath().toStdWString()).permissions() & std::filesystem::perms::owner_write);
+            if (readOnlyFolder) {
+                std::filesystem::permissions(fileInfoToDelete.absolutePath().toStdWString(), std::filesystem::perms::owner_write, std::filesystem::perm_options::add);
+            }
+            fakeFolder.localModifier().remove(file);
+            if (readOnlyFolder) {
+                std::filesystem::permissions(fileInfoToDelete.absolutePath().toStdWString(), std::filesystem::perms::owner_write, std::filesystem::perm_options::remove);
+            }
+        };
+
+        const auto renameReadOnly = [&] (const QString &relativePath, const QString &relativeDestinationDirectory)  {
+            auto sourceFileInfo = QFileInfo(fakeFolder.localPath() + relativePath);
+            auto destinationFileInfo = QFileInfo(fakeFolder.localPath() + relativeDestinationDirectory);
+            const auto readOnlySourceFolder = !static_cast<bool>(std::filesystem::status(sourceFileInfo.absolutePath().toStdWString()).permissions() & std::filesystem::perms::owner_write);
+            const auto readOnlyDestinationFolder = !static_cast<bool>(std::filesystem::status(destinationFileInfo.absolutePath().toStdWString()).permissions() & std::filesystem::perms::owner_write);
+            if (readOnlySourceFolder) {
+                std::filesystem::permissions(sourceFileInfo.absolutePath().toStdWString(), std::filesystem::perms::owner_write, std::filesystem::perm_options::add);
+            }
+            if (readOnlyDestinationFolder) {
+                std::filesystem::permissions(destinationFileInfo.absolutePath().toStdWString(), std::filesystem::perms::owner_write, std::filesystem::perm_options::add);
+            }
+            fakeFolder.localModifier().rename(relativePath, relativeDestinationDirectory);
+            if (readOnlySourceFolder) {
+                std::filesystem::permissions(sourceFileInfo.absolutePath().toStdWString(), std::filesystem::perms::owner_write, std::filesystem::perm_options::remove);
+            }
+            if (readOnlyDestinationFolder) {
+                std::filesystem::permissions(destinationFileInfo.absolutePath().toStdWString(), std::filesystem::perms::owner_write, std::filesystem::perm_options::remove);
+            }
+        };
+
+        const auto insertReadOnly = [&] (const QString &file, const int fileSize) {
+            auto fileInfo = QFileInfo(fakeFolder.localPath() + file);
+            const auto readOnlyFolder = !static_cast<bool>(std::filesystem::status(fileInfo.absolutePath().toStdWString()).permissions() & std::filesystem::perms::owner_write);
```

Comment:
@mgallien Why is this named `readOnlyFolder` while the input parameter is called `file`? You are checking the permissions of a file, not a folder here. I am confused.

Reply:
changed all occurrences to `isReadOnly`